py/persistentcode.c

```diff
@@ -398,7 +398,11 @@ mp_compiled_module_t mp_raw_code_load(mp_reader_t *reader, mp_module_context_t *
     if (MPY_FEATURE_DECODE_ARCH(header[2]) != MP_NATIVE_ARCH_NONE) {
         byte arch = MPY_FEATURE_DECODE_ARCH(header[2]);
         if (!MPY_FEATURE_ARCH_TEST(arch)) {
-            mp_raise_ValueError(MP_ERROR_TEXT("incompatible .mpy arch"));
+            if (MPY_FEATURE_ARCH_TEST(MP_NATIVE_ARCH_NONE)) {
+                mp_raise_ValueError(MP_ERROR_TEXT("MICROPY_EMIT_<ARCH> native support disabled"));
```

Comment:
I think it would be good to make the message more explicit about `.mpy`.  Eg "native code in .mpy not supported".

Reply:
Do you think it's helpful to still include `MICROPY_EMIT_<ARCH>` in the message, as a prompter of how to fix the issue?

Reply:
I think a comment is better than having that in the error message.